firebase-config/src/main/java/com/google/firebase/remoteconfig/internal/ConfigRealtimeHttpClient.java

```diff
@@ -277,7 +277,11 @@ public synchronized void retryHttpConnection() {
   }
 
   private synchronized void makeRealtimeHttpConnection(long retrySeconds) {
-    if (canMakeHttpStreamConnection() && httpRetriesRemaining > 0) {
+    if (!canMakeHttpStreamConnection()) {
+      return;
```

Comment:
Seems like we should return an error if there's a call to make the connection and it can't be made?